### PR TITLE
More FRB galaxies, including foreground galaxies

### PR DESCRIPTION
**Pull request by profxj**
A potpouri of code that was generated mainly to enable analysis of the galaxy
foreground to FRB 181112 (Prochaska et al. 2019, submitted to Science).

Here are some specific changes:

- Add RM and associated methods
- Add DM_ISM
- Add FGGalaxy object
- Update halo mass calculations for the new Aemulus HMF
- Add Virgo halo
- Simple Milky Way methods for DM

### CODE REVIEW

**Comment by profxj** [comment]
@caseyjlaw -- You might want to take a quick once over on what is new here.
I hope it doesn't (negatively) affect any of the RealFast bits and pieces.

File: frb/tests/test_frbtscatt.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def test_temp_smearing():
     tau = turb.temporal_smearing(lobs, zsource)
     # Test
     assert tau.unit == u.ms
```
[the remark this answers is not in the document]
Comment (profxj) [reply frb/tests/test_frbtscatt.py]: It is testing both that the variable is a `Quantity`
and that its 'default' unit is `ms`.

File: frb/tests/test_frbtscatt.py [hunk 2]
```diff
@@ -58,6 +58,6 @@ def test_ang_broadening():
     zsource = 2.
     theta = turb.angular_broadening(lobs, zsource)
     assert theta.unit == u.arcsec
```
[the remark this answers is not in the document]
Comment (profxj) [reply frb/tests/test_frbtscatt.py]: similar to the above.

File: frb/igm.py [hunk 3]
```diff
@@ -209,6 +209,7 @@ def avg_DMhalos(z, logMmin=10., f_diffuse=0.75, cumul=False):
           Redshift of the FRB
         logMmin: float, optional
           Lowest mass halos to consider
+          Cannot be much below 10.3 or the Halo code barfs
```
[the remark this answers is not in the document]
Comment (profxj) [reply frb/igm.py]: Msun;  doc string updated

**Comment by caseyjlaw** [comment]
I *think* there are no issues for me in this, although it is difficult to see when there are changes related to galaxy modeling and surveys and more.
Do you think pulling the surveys directory out as its own package is warranted? That may mean you wouldn't need to worry about how catalog querying tools like mine are affected by your work modeling galaxies, etc.

File: frb/tests/test_frbtscatt.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (profxj) [reply frb/tests/test_frbtscatt.py]: It is there to insist the units are `ms`

File: frb/igm.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (profxj) [reply frb/igm.py]: Added a comment.

**Comment by profxj** [comment]
Merging.

**Pull request merged**